src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -57,13 +63,24 @@ def pileupSizeTask(self):
             rucioAuthUrl = self.rucioClient.rucioParams['auth_host']
             rucioHostUrl = self.rucioClient.rucioParams['rucio_host']
             rucioToken, _tokenValidity = getRucioToken(rucioAuthUrl, self.rucioAccount)
-            containers = [r['pileupName'] for r in docs]
+            rucioScope = self.rucioScope
+            containers = []
+            for doc in docs:
+                pileupName = doc.get('customName', '')
+                if pileupName == '':
+                    pileupName = doc['pileupName']
+                else:
+                    rucioScope = spec.get('customRucioScope', 'group.wmcore')
+                containers.append(pileupName)
             msg = f"Fetching pileup size for {len(containers)} containers "
             msg += f"against rucio url: {rucioHostUrl}"
             self.logger.info(msg)
-            datasetSizes = getPileupContainerSizesRucio(containers, rucioHostUrl, rucioToken)
+            datasetSizes = getPileupContainerSizesRucio(containers, rucioHostUrl, rucioToken, scope=rucioScope)
```

Comment:
This logic is still wrong. You are passing a list of container names and associating ALL of them to the same scope, which is not going to be always true.

One potential way to resolve it is to call `getPileupContainerSizesRucio` twice - if needed. First with a list of containers under the `cms` scope, then a second call with a list of containers under the `group.wmcore` scope.

Reply:
I fixed the logic in a different way to avoid duplication rucio calls. Please see new code.